### PR TITLE
Add filter for Template Categories

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Add filter for Template Categories

Extract categories from Store Products result
Hide filter or categories if missing

[stage-2]

## Motivation and Context
Allow users to filter templates by various categories

## How Has This Been Tested?
Validated the changes locally. At this point no other categories are available other than Industry, which should not be a filter category. However, I added it for demo purposes until the other categories will be available. Will remove prior to merge.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
To confirm, I'll be removing the Industry filtering prior to release. That should hide the whole dropdown until the other categories are available.

@ezequielc please review. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc updated as discussed. 

Now the category picked only filters by the specific field - ie. picking `Library` will only filter through `templateLocations`.

Also, since there's some data in those fields, I've removed the Industry filtering so this is ready to go.

However, the data is not yet complete so my suggestion is to hold the release until Creative finalize their updates.

Thanks.

File: web/scripts/editor/filters/ftr-template-category.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+angular.module('risevision.editor.filters')
+  .filter('templateCategory', [
+    function () {
+      return function (templates, category) {
+        var categoryArrays = _.map(templates, category);
+        var categories = [].concat.apply([], categoryArrays);
+        var uniqueCategories = _.uniq(categories);
```
Comment (ezequielc) [inline web/scripts/editor/filters/ftr-template-category.js]: @alex-deaconu  We also need to sort the categories alphabetically according to the JTBD. Thanks!

Comment (alex-deaconu) [reply web/scripts/editor/filters/ftr-template-category.js]: @ezequielc good catch, updated! Thanks.

Comment (ezequielc) [reply web/scripts/editor/filters/ftr-template-category.js]: Looks great! Thanks

**Pull request merged**